File: Fase 2/Evidencias Proyecto/proyecto/app ionic/TatasApp/src/app/interfaces/familiar.ts
//interfaces para comparacion de contactos y usuarios tipo familiar registrados
//creado por ale el 25/04

//interfaces para poder comparar los contactos con los usuarios familiares registrados
export interface FamiliarRegistrado {
    id_usuario: number;
    telefono: string;
}

export interface ContactoAPI {
    nombre: string;
    telefono: string;
}

export interface RespuestaContactosAPI extends FamiliarRegistrado {
    contactos: ContactoAPI[];
}

//interface para registar un familiar
//creado por david el 02/05
export interface FamiliarRegistro {
    idAdultoMayor: number;
    idFamiliar: number;
}

//interfaces para obtener a los familiares registrados por el adulto mayor
//creado por david el 07/05
export interface FamiliarRelacionado {
    id_usuario: number;
    nombres: string;
    apellidos: string;
    correo: string;
    telefono: string;
    foto_perfil: string | null;
}

export interface FamiliarRegistrado {
    id_adulto_mayor: number;
    familiar_rel: FamiliarRelacionado;
}

File: Fase 2/Evidencias Proyecto/proyecto/api fastapi/app/schemas/familiar.py
#esquemas de familiar
#creado por david el 05/05

from pydantic import BaseModel
from typing import Optional


#REGISTRAR FAMILIARES

#esquema para guardar familiares
#creado por david el 25/04
class FamiliarCreate(BaseModel):
    adulto_mayor_id: int
    familiar_id: int

#########################################################################################

#OBTENER INFO DE FAMILIARES

#esquemas para obtener los familiares registrados en el grupo familiar del adulto mayor
#creado por david el 03/05
class UsuarioFamiliarOut(BaseModel):
    id_usuario: int
    nombres: str
    apellidos: str
    correo: str
    telefono: str
    foto_perfil: Optional[str]

    class Config:
        from_attributes = True

class FamiliarOut(BaseModel):
    id_adulto_mayor: int
    familiar_rel: UsuarioFamiliarOut

    class Config:
        from_attributes = True